test/e2e/pvc_test.go

```diff
@@ -0,0 +1,255 @@
+			t.Logf("PersistentVolume %s created", pvName)
+			name = pvName
+			break
+		}
+
+		if errors.IsAlreadyExists(err) {
+			continue
+		}
+
+		t.Logf("unable to create PersistentVolume %s: %s", pvName, err)
+	}
+
+	if name == "" {
+		return fmt.Errorf("unable to create PersistentVolume")
+	}
+
+	err := wait.Poll(1*time.Second, framework.AsyncOperationTimeout, func() (bool, error) {
+		pv, pvErr := client.PersistentVolumes().Get(name, metav1.GetOptions{})
+		return (pv != nil && pv.Status.Phase != corev1.VolumePending), pvErr
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createPVWithStorageClass(t *testing.T) error {
+	client := framework.MustNewClientset(t, nil)
+
+	storageClassList, err := client.StorageClasses().List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to list storage classes: %s", err)
+	}
+
+	for _, storageClass := range storageClassList.Items {
+		if err := createPV(t, storageClass.Name); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createPVC(t *testing.T, name string) error {
+	client := framework.MustNewClientset(t, nil)
+
+	claim := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: imageregistryv1.ImageRegistryOperatorNamespace,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteMany,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+
+	_, err := client.PersistentVolumeClaims(imageregistryv1.ImageRegistryOperatorNamespace).Create(claim)
+	if err != nil {
+		return err
+	}
+
+	err = wait.Poll(1*time.Second, framework.AsyncOperationTimeout, func() (bool, error) {
+		pvc, pvcErr := client.PersistentVolumeClaims(imageregistryv1.ImageRegistryOperatorNamespace).Get(name, metav1.GetOptions{})
+		return (pvc != nil && pvc.Status.Phase != corev1.ClaimPending), pvcErr
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkTestResult(t *testing.T, client *framework.Clientset) {
+	framework.MustEnsureImageRegistryIsAvailable(t, client)
+	framework.MustEnsureInternalRegistryHostnameIsSet(t, client)
+	framework.MustEnsureClusterOperatorStatusIsSet(t, client)
+	framework.MustEnsureOperatorIsNotHotLooping(t, client)
+
+	deploy, err := client.Deployments(imageregistryv1.ImageRegistryOperatorNamespace).Get(imageregistryv1.ImageRegistryName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if deploy.Status.AvailableReplicas == 0 {
+		framework.DumpYAML(t, "deployment", deploy)
+		t.Errorf("error: the deployment doesn't have available replicas")
+	}
+}
+
+func TestDefaultPVC(t *testing.T) {
+	client := framework.MustNewClientset(t, nil)
+
+	defer testDefer(t, client)
+
+	if err := createPV(t, ""); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := createPVWithStorageClass(t); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
why do we need to do both createPV and createPVWithStorageClass ?

Reply:
@bparees I need it to run locally without provisioner.

Reply:
In addition, if at some point someone turns provisioner off, then I want the test to work without storageclass.

Reply:
> @bparees I need it to run locally without provisioner.

why not just run it that way always then?

(if the autoprovisioner were working, then i think it would be reasonable to prefer to run the test using autoprovisioned PVs where possible, especially since those PVs will actually work, unlike the hostvolume ones you are creating which aren't actually backed by anything, but since you're creating the no-storage-class PVs manually either way, I don't see what purpose creating the ones with the storageclass serves?)

I think this comes back to: you should talk to the storage team to learn why PVs aren't being autoprovisioned on AWS.

Reply:
> @bparees Because when I create a PVC in a cluster, the admission plugin automatically adds a storageclass to it and my PV without a storageclass does not bound. I can't control it. That is why I create all kinds of PV (with and without storageclass).

thanks, that helps (I do recall you mentioning part of this issue on scrum, now that you bring it up again here).

But i take it you've seen some clusters (your own local ones) that do not have the admission plugin behavior setting a storageclass on the PVC.

Putting a couple comments in the code explaining why you're creating 2 PVs/what you've observed would be helpful for the next person who's looking at this.

Reply:
> But i take it you've seen some clusters (your own local ones) that do not have the admission plugin behavior setting a storageclass on the PVC.

@bparees For example `oc cluster up` (v3.11).

Reply:
wow.  that is brave/dedicated to keep making oc cluster up work.

Reply:
> I'm pretty sure we don't have a autoprovisioner. Only admission plugin.

I was wrong. We have an autoprovisioner there, but for AWS. AWSElasticBlockStore do not support ReadWriteMany access mode [1].

[1] https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes